mods/tnt/init.lua

```diff
@@ -384,7 +349,74 @@ minetest.register_craft({
 	}
 })
 
+function tnt.register_tnt(def)
+	local name = ""
+	if not def.name:find(':') then
+		name = "tnt:"..def.name
+	else
+		name = def.name
+		def.name = def.name:sub(def.name:find(':')+1)
```

Comment:
Matter of style, reliability and likely more performant since you use only one Lua method.